core/src/main/java/hudson/TcpSlaveAgentListener.java

```diff
@@ -131,6 +133,18 @@ public int getAdvertisedPort() {
         return CLI_PORT != null ? CLI_PORT : getPort();
     }
 
+    /**
+     * Gets the host name that we advertise protocol clients to connect to.
+     */
+    public String getAdvertisedHost() {
```

Comment:
Not sure if it's the best approach. 
Why don't unrestrict CLI_HOST_NAME or
Why don't just read the `TcpSlaveAgentListener.class.getName()+".hostName"` system property without having to rely on a newer version of the core.

Reply:
Unrestricting CLI_HOST_NAME would be OK. Still, getAdvertisedHost() additionally to getAdvertisedPort() sounds more intuitive to me.

[jenkins.util.SystemProperties](https://github.com/jenkinsci/jenkins/blob/master/core/src/main/java/jenkins/util/SystemProperties.java#L80) which is used in [TcpSlaveAgentListener](https://github.com/jenkinsci/jenkins/blob/master/core/src/main/java/hudson/TcpSlaveAgentListener.java#L540) is also restricted.

`System.getProperty(...)` is what @jglick proposed (and I [implemented](https://github.com/jenkinsci/kubernetes-plugin/pull/602/commits/91b7d13f6099bc22af48e4ad0ec16fde5325ab9d#diff-ec0e1841b0d5d888d11370d1a10eeabaR347)) in the Kubernetes plugin pull request, but [jenkins.util.SystemProperties.getString(...)](https://github.com/jenkinsci/jenkins/blob/master/core/src/main/java/jenkins/util/SystemProperties.java#L184) is doing more.